testing/utils.py

```diff
@@ -0,0 +1,52 @@
+import os
+from contextlib import contextmanager
+from collections import Mapping
+
+
+@contextmanager
+def cd(path):
```

Comment:
All of this fancy stuff came from you right?  Very swanky

Reply:
it must have come from somewhere in the first place, but it ended up in my toolbox as it's a very useful piece of code :)

Reply:
haha, just realised it was used in sphinx as well: ``from sphinx.util.osutil import cd``